### PR TITLE
Features/social account [OSF-6597]

### PR DESCRIPTION
**Pull request by jnayak1**
Authentication through [django allauth](https://github.com/pennersr/django-allauth). Allows users to sign in with their osf account through oauth and it creates a user on our django app for them.

New packages require installation:
pip install -r requirements.txt

Workflow:
1. Go to localhost:8000/accounts/login
2. Click Open Science Framework
3. Login to OSF
4. Allow permission
5. Redirected back to osf-meetings homepage (localhost:4200)

Ticket:
https://openscience.atlassian.net/browse/OSF-6597

### CODE REVIEW

File: meetings/osf_oauth2_adapter/apps.py [hunk 1]
```diff
@@ -0,0 +1,12 @@
+from __future__ import unicode_literals
+
+from django.apps import AppConfig
+import os
+
+class OsfOauth2AdapterConfig(AppConfig):
+    name = 'osf_oauth2_adapter'
+    # staging by default so people don't have to run OSF to use this.
+    osf_api_url = os.environ.get('OSF_API_URL', 'https://staging-api.osf.io').rstrip('/') + '/'
+    osf_accounts_url = os.environ.get('OSF_ACCOUNTS_URL', 'https://staging-accounts.osf.io').rstrip('/') + '/'
```
Comment (Ghalko) [inline meetings/osf_oauth2_adapter/apps.py:10]: We might not want to hardcode this.

File: meetings/osf_oauth2_adapter/migrations/0001_make_human_group.py [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+    Group.objects.create(name=OsfOauth2AdapterConfig.humans_group_name)
+
+
+class Migration(migrations.Migration):
+
+    dependencies = [
+    ]
+
+    operations = [
+        migrations.RunPython(create_human_group),
```
Comment (Ghalko) [inline meetings/osf_oauth2_adapter/migrations/0001_make_human_group.py]: The migrations could be combined.

File: meetings/osf_oauth2_adapter/apps.py (hunk 1 above)
Comment (jnayak1) [inline meetings/osf_oauth2_adapter/apps.py]: I copied this code from osf [SHARE](https://github.com/CenterForOpenScience/SHARE/blob/develop/osf_oauth2_adapter/apps.py). The idea of it is to use staging by default but we can specify a different server with OSF_API_URL, and OSF_ACCOUNTS_URL.

**Comment by Ghalko** [comment]
Looks good.

**Pull request merged**